firecrown/ccl/sources/sources.py

```diff
@@ -374,3 +374,80 @@ def render(self, cosmo, params, systematics=None):
             systematics[systematic].apply(cosmo, params, self)
 
         self.dndz_interp_ = Akima1DInterpolator(self.z_, self.dndz_)
+
+
+class CMBLSource(Source):
+    """A CCL CMB Lensing Source.
+
+    Parameters
+    ----------
+    sacc_tracer : str
+        The name of the tracer in the SACC file.
+    scale : float, optional
+        The default scale for this source. Usually the default of 1.0 is
+        correct.
+    systematics : list of str, optional
+        A list of the source-level systematics to apply to the source. The
+        default of `None` implies no systematics.
+
+    Attributes
+    ----------
+    beam_orig : np.ndarray
+        The original beam used in the sacc tracer before any systematics
+        are applied. Set after a call to `read`.
+    beam_: np.ndarray
+        The beam used for CMB lensing. Set after a call to `render`.
+    scale_ : float
+        The overall scale associated with the source. Set after a call to
+        `render`.
+    tracer_ : `pyccl.WeakLensingTracer`
+        The CCL tracer associated with this source. Set after a call to
+        `render`.
+
+    Methods
+    -------
+    render : apply systematics to this source and build the
+        `pyccl.CMBLSource`
+    """
+    def __init__(self, *, sacc_tracer, scale=1.0, systematics=None):
+        self.sacc_tracer = sacc_tracer
+        self.scale = scale
+        self.systematics = systematics or []
+
+    def read(self, sacc_data):
+        """
+        Read the data for this source from the SACC file.
+
+        Parameters
+        ----------
+        sacc_data : sacc.Sacc
+            The data in the sacc format.
+        """
+        tracer = sacc_data.get_tracer(self.sacc_tracer)
+        beam = tracer.beam.copy().flatten()
+        self.beam_orig = beam
+
+    def render(self, cosmo, params, systematics=None):
+        """
+        Render a source by applying systematics.
+
+        Parameters
+        ----------
+        cosmo : pyccl.Cosmology
+            A pyccl.Cosmology object.
+        params : dict
+            A dictionary mapping parameter names to their current values.
+        systematics : dict
+            A dictionary mapping systematic names to their objects. The
+            default of `None` corresponds to no systematics.
+        """
+        systematics = systematics or {}
+
+        self.beam_ = self.beam_orig.copy()
```

Comment:
Not sure the beam is being used for anything, actually. In any case, it's unlikely that the CMB lensing maps will be provided with a beam (these usually come from beam-deconvolved CMB maps). They might come with a noise model (e.g. some kind of N_ell), but that's not necessarily relevant here unless you care about the CMB lensing auto-correlation or unless you're generating covariances.

Reply:
Ok, Thanks. I'm not planning to do anything like auto-correlation or covariances here. So I'll just remove the beam?

Reply:
Sounds good